src/modules/returns/lib/csv-adapter/date-parser.js

```diff
@@ -51,7 +52,8 @@ const getDateFrequency = date => {
     const m = parsers[timePeriod](date)
     return m.isValid() ? { timePeriod, moment: m } : null
   })
-  return compact(result)[0]
+
+  return result.filter(x => x)[0]
```

Comment:
Does `result.first(x => x)` work here?

Reply:
Changing this to .first doesn't work as its complaining its not a function `  result.first is not a function` 😢

Reply:
Sorry my mistake! I meant `result.find(x => x)` 🙄